### PR TITLE
Return query status for statistic executor

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：
for https://github.com/StarRocks/starrocks/issues/4635

1. Directly return error when column cardinality exceed 256
2. Support return Exec Status for statistic sql, mainly change is register and unregister query id
3. Remove asyncLoadAll method

### CODE REVIEW

File: be/src/storage/vectorized/meta_reader.cpp [hunk 1]
```diff
@@ -292,9 +292,7 @@ Status SegmentMetaCollecter::_collect_dict(ColumnId cid, vectorized::Column* col
 
     std::vector<Slice> words;
     if (!_column_iterators[cid]->all_page_dict_encoded()) {
-        // if all_page_dict_encoded if false, return fake dict word which cardinality exceed low cardinality base size
-        // so FE will not collect again and mark this column not a low cardinality
-        words = FAKE_DICT_SLICE_WORDS;
```
Comment (laotan332) [inline be/src/storage/vectorized/meta_reader.cpp:297]: remove code about FAKE_DICT_SLICE_WORDS ?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 10 / 29 (34.48%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CacheDictManager.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|10|19|52.63%|

**Review by stdpain — DISMISSED** [review]

File: be/src/storage/vectorized/meta_reader.cpp (hunk 1 above)
Comment (kangkaisen) [reply be/src/storage/vectorized/meta_reader.cpp:297]: done

**Review by stdpain — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 29 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CacheDictManager.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|19|00.00%|

**Pull request merged**